File: treenode.java
import javax.swing.*;
import javax.swing.tree.DefaultMutableTreeNode;

public class treenode extends JFrame {
    JTree tree;
    JLabel label;

    public treenode() {
        // Create root and child nodes
        DefaultMutableTreeNode root = new DefaultMutableTreeNode("B");
        DefaultMutableTreeNode n1 = new DefaultMutableTreeNode("E");
        DefaultMutableTreeNode n2 = new DefaultMutableTreeNode("R");
        DefaultMutableTreeNode n3 = new DefaultMutableTreeNode("O");

        // Add child nodes to the root
        root.add(n1);
        root.add(n2);
        root.add(n3);

        // Adding children to n1, n2, and n3
        DefaultMutableTreeNode n1Child = new DefaultMutableTreeNode("C");
        n1.add(n1Child);  

        DefaultMutableTreeNode n2Child = new DefaultMutableTreeNode("S");
        n2.add(n2Child);  

        DefaultMutableTreeNode n3Child = new DefaultMutableTreeNode("A");
        n3.add(n3Child);  

        // Create the tree
        tree = new JTree(root);
        add(tree);

        // Get the child count for each node
        int childCount = root.getChildCount();
        int childrenCountn1 = n1.getChildCount();
        int childrenCountn2 = n2.getChildCount();
        int childrenCountn3 = n3.getChildCount();

        // Corrected the HTML string to refer to the correct node names
        label = new JLabel("<html>The root node has " + childCount + " children.<br>" +
                           "n1 (E) has " + childrenCountn1 + " children.<br>" +
                           "n2 (R) has " + childrenCountn2 + " children.<br>" +
                           "n3 (O) has " + childrenCountn3 + " children.</html>");

        // Add the label to the frame
        add(label, "North");

        // Set up the JFrame window
        this.setTitle("JTree Example");
        this.setSize(300, 300);
        this.setVisible(true);
    }

    public static void main(String[] args) {
        new treenode();
    }
}

File: treenode.py
import tkinter as tk
from tkinter import ttk

class TreeNodeApp(tk.Tk):
    def __init__(self):
        super().__init__()

        self.title("TreeNode Example")
        self.geometry("300x300")

        # Create a treeview widget
        self.tree = ttk.Treeview(self)
        self.tree.pack(expand=True, fill=tk.BOTH)

        # Add root node
        root = self.tree.insert("", "end", "B", open=True)

        # Add child nodes to the root
        n1 = self.tree.insert(root, "end", "E", open=True)
        n2 = self.tree.insert(root, "end", "R", open=True)
        n3 = self.tree.insert(root, "end", "O", open=True)

        # Add children to n1, n2, and n3
        self.tree.insert(n1, "end", "C")
        self.tree.insert(n2, "end", "S")
        self.tree.insert(n3, "end", "A")

        # Get the child count for each node
        childCount = len(self.tree.get_children(root))
        childrenCountn1 = len(self.tree.get_children(n1))
        childrenCountn2 = len(self.tree.get_children(n2))
        childrenCountn3 = len(self.tree.get_children(n3))

        # Display the child counts in a label
        label_text = f"The root node has {childCount} children.\n" \
                     f"n1 (E) has {childrenCountn1} children.\n" \
                     f"n2 (R) has {childrenCountn2} children.\n" \
                     f"n3 (O) has {childrenCountn3} children."
        
        label = tk.Label(self, text=label_text)
        label.pack(side=tk.TOP, padx=10, pady=10)

if __name__ == "__main__":
    app = TreeNodeApp()
    app.mainloop()
